mne/bem.py

```diff
@@ -0,0 +1,321 @@
+# Authors: Alexandre Gramfort <alexandre.gramfort@telecom-paristech.fr>
+#          Matti Hamalainen <msh@nmr.mgh.harvard.edu>
+#          Eric Larson <larson.eric.d@gmail.com>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+from scipy import linalg, optimize
+
+from .fixes import partial
+from .utils import verbose, logger
+from .io.constants import FIFF
+from .externals.six import string_types
+
+
+# ############################################################################
+# Compute EEG sphere model
+
+def _fwd_eeg_get_multi_sphere_model_coeffs(m, n_terms):
+    """Get the model depended weighting factor for n"""
+    nlayer = len(m['layers'])
+    if nlayer in (0, 1):
+        return 1.
+
+    # Initialize the arrays
+    c1 = np.zeros(nlayer - 1)
+    c2 = np.zeros(nlayer - 1)
+    cr = np.zeros(nlayer - 1)
+    cr_mult = np.zeros(nlayer - 1)
+    for k in range(nlayer - 1):
+        c1[k] = m['layers'][k]['sigma'] / m['layers'][k + 1]['sigma']
+        c2[k] = c1[k] - 1.0
+        cr_mult[k] = m['layers'][k]['rel_rad']
+        cr[k] = cr_mult[k]
+        cr_mult[k] *= cr_mult[k]
+
+    coeffs = np.zeros(n_terms - 1)
+    for n in range(1, n_terms):
+        # Increment the radius coefficients
+        for k in range(nlayer - 1):
+            cr[k] *= cr_mult[k]
+
+        # Multiply the matrices
+        M = np.eye(2)
+        n1 = n + 1.0
+        for k in range(nlayer - 2, -1, -1):
+            M = np.dot([[n + n1 * c1[k], n1 * c2[k] / cr[k]],
+                        [n * c2[k] * cr[k], n1 + n * c1[k]]], M)
+        num = n * (2.0 * n + 1.0) ** (nlayer - 1)
+        coeffs[n - 1] = num / (n * M[1, 1] + n1 * M[1, 0])
+    return coeffs
+
+
+def _compose_linear_fitting_data(mu, u):
+    # y is the data to be fitted (nterms-1 x 1)
+    # M is the model matrix      (nterms-1 x nfit-1)
+    for k in range(u['nterms'] - 1):
+        k1 = k + 1
+        mu1n = np.power(mu[0], k1)
+        u['y'][k] = u['w'][k] * (u['fn'][k1] - mu1n*u['fn'][0])
```

Comment:
Travis automatically checks PEP8 now, so there really shouldn't be an error, what did you have in mind?